nn_meter/dataset/gnn_dataloader.py

```diff
@@ -1,2 +1,274 @@
 # Copyright (c) Microsoft Corporation.
-# Licensed under the MIT license.
\ No newline at end of file
+# Licensed under the MIT license.
+import torch
+import jsonlines
+from .bench_dataset import bench_dataset
+import os
+import random
+import dgl
+
+RAW_DATA_URL = "https://github.com/microsoft/nn-Meter/releases/download/v1.0-data/datasets.zip"
+__user_dataset_folder__ = os.path.expanduser('~/.nn_meter/dataset')
+
+hws = [
+    "cortexA76cpu_tflite21",
+    "adreno640gpu_tflite21",
+    "adreno630gpu_tflite21",
+    "myriadvpu_openvino2019r2",
+]
+
+
+class GNNDataset(torch.utils.data.Dataset):
+    def __init__(self, data_dir=__user_dataset_folder__, train=True, device="cortexA76cpu_tflite21", split_ratio=0.8):
+        """
+        Dataloader of the Latency Dataset
+
+        Parameters
+        ----------
+        data_dir : string
+            Path to save the downloaded dataset
+        train: bool
+            Get the train dataset or the test dataset
+        device: string
+            The Device type of the corresponding latency
+        shuffle: bool
+            If shuffle the dataset at the begining of an epoch
+        batch_size: int
+            Batch size.
+        split_ratio: float
+            The ratio to split the train dataset and the test dataset.
+        """
+        err_str = "Not supported device type"
+        assert device in hws, err_str
+        self.device = device
+        self.data_dir = data_dir
+        self.train = train
+        self.split_ratio = split_ratio
+        self.adjs = {}
+        self.attrs = {}
+        self.nodename2id = {}
+        self.id2nodename = {}
+        self.op_types = set()
+        self.opname2id = {}
+        self.raw_data = {}
+        self.name_list = []
+        self.latencies = {}
+        self.download_data()
```

Comment:
I suggest using `self.data_dir = self.download_data()` and return the dataset path in `self.download_data`.